spec/foundation.md

```diff
@@ -0,0 +1,615 @@
+
+escape-char = %x1B
+new-line = %x0A
+```
+
+A message is **fenced** when it is preceded by one ESC character and succeeded by another ESC character, followed by a newline.
+Fencing is only used for messages without client ID (see section 3.2).
+
+*Rationale:* The use of ESC characters as delimiters is inspired by ANSI escape sequences, which also start with an ESC character.
+The trailing `\n` is considered part of the fenced message because, in section 3.2.2, we recommend that messages be placed on separate lines to simplify the job of line-based text filters such as `grep`, `sed` and `awk`.
+If the message did not end with a `\n` by itself, it could not be placed on its own line without introducing a superfluous line break in the surrounding text data.
+
+#### 3.1.3. Human-readable representation
+
+The following alternative representation for netstrings and messages may be used when showing messages in specifications or informational displays, such as logs:
+
+- In a message with client ID, the first netstring (the client ID) is represented by its value, enclosed in angle brackets (`<` and `>`).
+- Netstrings whose value matches the regular expression `^[A-Za-z0-9._-]*$` are represented directly by their value.
+- Other netstrings are represented as C string literals with the same value.
+- Messages are represented by a whitespace-separated concatenation of the representations of the netstrings contained in it, enclosed in parentheses.
+
+*Rationale:* This format is more human-readable and thus better suited for examples within specifications, or for diagnostic output of programs that process VT6 messages.
+
+For example:
+
+```
+message without client ID:
+
+actual         = {3|8:core.set,13:example.title,13:hello "world",}
+human-readable = (core.set example.title "hello \"world\"")
+
+message with client ID:
+
+actual         = {4:a1b2,3|8:core.set,13:example.title,13:hello "world",}
+human-readable = (<a1b2> core.set example.title "hello \"world\"")
+```
+
+### 3.2. Using the standard streams
+
+#### 3.2.1. Standard input
+
+When a VT6 client reads from stdin, it SHOULD look for events, fenced messages that are sent by its producing peer.
+
+Once it reads the byte sequence `<ESC>{` (decimal byte sequence 27, 123) from its stdin, it SHALL attempt to read an entire fenced message.
+If, upon further reading, the initial `<ESC>{` turns out not to start a syntactically valid fenced message, those bytes, as well as all bytes up to the next `<ESC>{` SHALL be treated as ordinary input data.
+If, however, a syntactically valid fenced message is read from stdin, it shall be processed as a single unit occurring within the input stream, rather than as a string of individual bytes.
+The bytes that make up the fenced message SHALL NOT be considered part of the input text.
+
+When a VT6 terminal (or an output proxy) reads a client's stdout, it MUST look for events in the same way as described above.
```

Comment:
Doesn't the case for clients also hold for output proxies? Aren't they also just clients? Isn't anything a client that is not a terminal?

Reply:
Yes. But the case for output proxies is stronger because it is a MUST instead of a SHOULD.